### ISSUE TITLE
high level simulator

### ISSUE THREAD

**Issue by JAndringa** [RoboTeamTwente/roboteam#30]
**Is your feature request related to a problem? Please describe.**
grSim is a very slow and time based low level simulator. This makes it very slow to test play decision making and would make machine learning very difficult.

**Describe the solution you'd like**
A high level simulator would enable us to make it frame based and also way faster. This will in turn enable us to do easy machine learning.

**Describe alternatives you've considered**
The ideal way would be to just use the Tigers simulator, but given their definition of open source, this will also be very difficult.

**Comment by rolfvdhulst** [comment]
@JAndringa Why is it difficult to use the Tigers simulator? I believe they literally open sourced their code of previous years, including their simulator, on their webpage?
Regardless, it may be easier/better to simply write your own (in C++ rather than Java), as this will be a lot faster. Feel free to ask for help with the physics if you need it.

**Comment by JAndringa** [comment]
as far as I know (and could find) Tigers has only open sourced their auto referee. Their simulator is built into their framework so that we can only use it with their AI and not ours.

**Comment by rolfvdhulst** [comment]
Yeah, their simulator is built into their framework indeed, but the source code of their framework can be found here;
https://github.com/TIGERs-Mannheim/Sumatra/tree/master/modules/moduli-vision-simulation. Although it is Java, this might help a bit with making some decisions about how to simulate things.
